### ISSUE TITLE
AmiseBW for non-int data

### ISSUE THREAD

**Issue by darrencl** [rigdenlab/conkit#95]
Hi, it seems like the Bandwidth function, especially `AmiseBW` only accept integer data? I have my data in `float` and get below error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-279-9f6c160fda49> in <module>
      1 from conkit.misc.bandwidth import AmiseBW
      2 
----> 3 AmiseBW(X).bandwidth

~/anaconda3/lib/python3.7/site-packages/conkit/misc/bandwidth.py in bandwidth(self)
     77         data = self._data
     78         x0 = BowmanBW(data).bandwidth
---> 79         y0 = c_optimize_bandwidth(data, x0)
     80         x = 0.8 * x0
     81         y = c_optimize_bandwidth(data, x)

~/anaconda3/lib/python3.7/site-packages/conkit/misc/ext/c_bandwidth.pyx in conkit.misc.ext.c_bandwidth.c_optimize_bandwidth()

ValueError: Buffer dtype mismatch, expected 'int64_t' but got 'double'
```

**Comment by FilomenoSanchez** [comment]
Hi @darrencl , yes I am afraid that the current implementation of this function expects integers, not floats (it's written in cython so you need to specify types for the input variables). This is because we use this function to calculate the density of contacts along the sequence of residues, so it makes sense to expect integers. If you want this to work with doubles, I suggest the following changes:
```diff
diff --git a/conkit/misc/ext/c_bandwidth.pyx b/conkit/misc/ext/c_bandwidth.pyx
index c5a046a..3a9877f 100644
--- a/conkit/misc/ext/c_bandwidth.pyx
+++ b/conkit/misc/ext/c_bandwidth.pyx
@@ -12,7 +12,7 @@ cdef double SQRT_PI = sqrt(M_PI)
 cdef double SQRT_2PI = sqrt(2.0 * M_PI)
 
 
-def c_optimize_bandwidth(np.ndarray[np.int64_t, ndim=2] A, double v):
+def c_optimize_bandwidth(np.ndarray[np.float64, ndim=2] A, double v):
     cdef double alpha, sigma, integral
     alpha = 1.0 / (2.0 * SQRT_PI)
     sigma = 1.0
@@ -20,7 +20,7 @@ def c_optimize_bandwidth(np.ndarray[np.int64_t, ndim=2] A, double v):
     return v - ((A.shape[0] * integral * sigma**4) / alpha)**(-1.0 / (A.shape[1] + 4))
 
 
-def c_get_stiffness_integral(np.ndarray[np.int64_t, ndim=2] A, double v, double eps):
+def c_get_stiffness_integral(np.ndarray[np.float64, ndim=2] A, double v, double eps):
     cdef Py_ssize_t i, j, n
     cdef double min_, max_, dx, maxn, yy, y1, y2, y3, y
     min_ = A.min() - v * 3
@@ -48,7 +48,7 @@ def c_get_stiffness_integral(np.ndarray[np.int64_t, ndim=2] A, double v, double
     return yy
 
 
-def c_get_gauss_curvature(np.ndarray[np.int64_t, ndim=2] A, double x, double w):
+def c_get_gauss_curvature(np.ndarray[np.float64, ndim=2] A, double x, double w):
     cdef Py_ssize_t i, j
     cdef double w_sq, w_sqrt_2pi, curvature, z
     w_sq = w*w
```
Basically you have to change `np.int64_t` with `np.float64` (or `np.float128` depending on the nature of your problem). However please note that I did not write this function so I'm not sure what the consequences will be, and depending on what you want to do this might not be the only change you will have to do. Also please note that the expected input would be an array like this (can be found at the tests in `conkit/misc/tests/test_bandwidth.py`):

```python
>>> xy = np.array([(1, 5), (3, 3), (2, 4)], dtype=np.int64) 
>>> x = np.asarray([i for (x, y) in xy for i in np.arange(x, y + 1)])[:, np.newaxis]
>>> print(x)
array([[1],
       [2],
       [3],
       [4],
       [5],
       [3],
       [2],
       [3],
       [4]])

```

**Comment by FilomenoSanchez** [comment]
@darrencl I am closing this as it's been two days since I answered and it seems that this might have solved the question. Please feel free to re-open the issue if otherwise.
